crates/wysiwyg/src/dom/dom_methods.rs

```diff
@@ -434,6 +434,73 @@ where
         self.assert_invariants();
     }
 
+    /// Returns two new subtrees as the result of splitting the Dom symetrically without mutating
+    /// itself. Also returns the new handles of node that was split.
+    ///
+    /// Only returns nodes that are modified by the split and ignores any nodes which were not
+    /// either split or contain a node that was split.
+    pub(crate) fn split_new_sub_trees(
```

Comment:
This code makes sense, but take into account that it can be expensive memory-wise, as you're duplicating the whole Dom. 

In the quotes PR I've modified `split_sub_tree` to also take a `to_handle: Option<DomHandle>` so you can extract just part of the tree, not halves of it, so it might be a good idea to replace this function in the future with it, or modify the `split_aub_tree` function to return an actual `Dom<S>` object instead of a container, along with the `DomHandle` of the `from_handle` and `to_handle` inside of it if needed?

Reply:
> In the quotes PR I've modified `split_sub_tree` to also take a `to_handle: Option<DomHandle>` so you can extract just part of the tree, not halves of it, so it might be a good idea to replace this function in the future with it,

Thank you for letting me know about this, once the modified `split_sub_tree` method is here I'll update this one to use this.